### DISCUSSION TITLE
Does winget allow for repackaging?

### DISCUSSION THREAD

**Discussion by DarthXibalba** [microsoft/winget-pkgs#11378] [Q&A]
Let's say I wanted to take my favorite release of Docker from winget _(d/Docker/DockerDesktop/3.3.1.63152)_ can I repackage it as /d/DarthXibalbaRedist/Docker/DockerDesktop/v1/?

Granted the licensing permitted for this, is this kind of redistribution acceptable within the winget community?
What I'm trying to accomplish is to easily redistribute a concrete set of software packages for a targeted set of users who use those specific tools.

More info: When I say "repackage it" I mean uploading a manifest that would point to:
- a copy of the docker 3.3.1.63152 installer that I would maintain in some repository
OR
- the original publisher's manifest _(manifests/d/Docker/DockerDesktop/3.3.1.63152/Docker.DockerDesktop.yaml)_

Does winget support manifest files that are pointers to other manifest files?

**Comment by denelon** [comment]
@DarthXibalba this seems like a great candidate for a third party repository. For this repository we're looking for packages to come from the ISV/Publisher's source. There isn't a concept of a pointer manifest.

When we implement dependencies, one package could depend on another but I think that's a slightly different scenario than what you are asking for.

**Reply by DarthXibalba** [reply]
@denelon can you elaborate on what you mean by "third party repository"? Do you mean hosting my own winget-dx-packages github repo that would contain the software bundle I'm trying to redistribute?

How do dependencies work in winget? I've achieved a similar package redist mechanism using Ubuntu's apt-get where I can create a package that has no payload but has several other dependencies that get installed when installing my metapackage.

Ex: apt-get install darthxibalba-utils installs dependencies of vscode, git, python3, etc

Is this kind of functionality possible with winget?

**Reply by denelon** [reply]
We're implementing a REST API based "source". You could essentially host any packages you wanted.

Dependencies haven't been implemented yet, we've added support in the schema, but the implementation hasn't started yet. We're looking at four different types of dependencies: Windows libraries, packages from the same source, external dependencies, and Windows features.

The import feature allows you to specify multiple packages from multiple sources, but we don't currently have a concept of a meta-package yet.
